### ISSUE TITLE
weird behavior during substitutions

### ISSUE THREAD

**Issue by jmague** [jqlang/jq#1239]
Hi,

I've came across a case where substitutions clearly don't work as expected. 

```
$ echo '"“@a"' | jq -r 'sub("a";"A")'
Aa
```

Here are a couple of similar examples: 

```
$ echo '"@“a"' | jq -r 'sub("a";"A")'
A
```

or

```
$ echo '"“@ a bc\nd"' | jq -r 'sub("\n";" ";"g")'
“@ a  c d
```

I haven't figure out the regularity behind these errors, but the cooccurrence of _“@_ seems to be part of the problem.

**Comment by pkoppstein** [comment]
The problems you have found have been fixed since the release of jq 1.5.  Please use the "development" version of jq, or try a workaround.

**Comment by jmague** [comment]
I'm actually using version 1.5

$ jq --version && echo '"@ a bc\nd"' | jq -r 'sub("\n";" ";"g")'
jq-1.5
@ a bc d

**Comment by pkoppstein** [comment]
@jmague -

(1) As I said, the problem you described with `“` was resolved AFTER the release of v1.5. 

(2) The output you show using `"@ a bc\nd"` as input and `'sub("\n";" ";"g")'` as filter seems correct to me.  That is, the NEWLINE was replaced with a space character.  What were you expecting?

**Comment by itchyny** [comment]
This bug was already fixed and released in [1.6](https://github.com/jqlang/jq/releases/tag/jq-1.6) in Nov 2018, so closing this issue.
